### ISSUE TITLE
Interpret highway=pedestrian properly

### ISSUE THREAD

**Issue by dabreegster** [a-b-street/ltn#64]
https://www.openstreetmap.org/way/353926198
There should be filters on these by default

**Comment by XioNoX** [comment]
Thanks, came here to open that issue :)

**Comment by dabreegster** [comment]
![image](https://github.com/user-attachments/assets/e4f6b5e1-d614-4f75-a6b7-2708436fcb11)
This can't be done in a straightforward way of one (or two) filters per road segment with the tag, or this monstrosity occurs. The filters need to only get placed where these special case roads meet normal roads.

**Comment by dabreegster** [comment]
![image](https://github.com/user-attachments/assets/02ce6063-0863-4f19-8e70-2c4ae209e6a1)
Not perfect, but the Manchester example is looking more tame. Need to rethink the red disconnected cells -- this was meant to show user error originally, but they're pedestrianized areas here.

**Comment by XioNoX** [comment]
Thanks ! That works fine for my use-cases based on my quick tests.

The red areas should ideally show streets that are only reachable through a street with a different road restriction than they currently have (making them "disconnected"/"islands").

**Comment by dabreegster** [comment]
Manchester center https://www.openstreetmap.org/#map=16/53.47990/-2.24700 is a great complex test case. Ultimate goal is to handle these complex city centre pedestrianization schemes nicely, whatever that means. A next step to improve there is looking at adjacent cells that count as separate -- is that necessary? The existing filter logic uses different filter types right now for not clear reasons, and merging cells might be possible.

If there are other pedestrianized centre areas like Manchester, would be useful to have more (and simpler) test cases. @XioNoX has mentioned a few in different issues; mind linking to something like https://www.openstreetmap.org/#map=16/48.58186/7.74710 (which looks like an enormous island-ish thing that has no through traffic already?)

**Comment by dabreegster** [comment]
Ah, the Grande Île example isn't fully pedestrianized; streets on the north side seem open. Here's a quick view of it, is it roughly correct today? (Excluding the angry red cell that isn't really an error that it's disconnected):
![image](https://github.com/user-attachments/assets/6fa5ae73-4879-42bb-9ec4-6f4e370cf93b)
Or maybe the other styling is easier to see:
![image](https://github.com/user-attachments/assets/4b100edd-c07f-4982-a31b-eb67b9ffdaf4)

**Comment by TFCx** [comment]
From what I remember, it's mostly correct (but I lived there 15 yrs ago): the north part wasn't pedestrian, as well as the south part near the Gutemberg place.

As I saw your 2nd screenshot, I'm now wondering if when displaying color on roads, the inaccessible part should still be displayed in "block mode". It's somewhat usual in cartography to have the pedestrian areas / touristic centers displayed in another color. Inaccessible areas are still very specific in their functionality. A lighter red might also help with the visualization.

**Comment by dabreegster** [comment]
Another possible problem: we draw thick white lines to indicate an interior road with 0 shortcuts. The basemap uses dotted lines for footways.

![Image](https://github.com/user-attachments/assets/e858dd9c-6242-4515-a9c0-4d3be7798851)
vs

![Image](https://github.com/user-attachments/assets/8001419f-231f-48fd-841a-ba9ede865a84)

Not sure yet if this is confusing. #75 would help, showing that a road is marked pedestrian in OSM.

**Comment by TFCx** [comment]
Top picture was very intuitive for me. I thought you screenshot gmap / a real cartographic app. I'm wondering if unreachable roads have another meaning in real life other than pedestrian zones?

**Comment by dabreegster** [comment]
Top picture is just maptiler streets-v2 basemap :)

Unreachable cells either represent some kind of pre-existing pedestrian zone (using red doesn't make sense) or it could be a user error after placing too many filters and unintentionally disconnecting things

**Comment by TFCx** [comment]
> it could be a user error after placing too many filters and unintentionally disconnecting things

Yes, but in that case, it's really simple to understand that your last click put one too many filter. And it's very easy to undo that action (ctrl+Z or just left click again on the last spawned icon). The only concern I would have for this visualization trick is that it might be too abrupt but so is the coloring in red.

Could you test the thinning on these unzoomed views ? https://github.com/a-b-street/ltn/issues/64#issuecomment-2585784610

**Comment by dabreegster** [comment]
What does "thinning" mean? If I take away the special red case and show it like any other cell, it's like this:

![Image](https://github.com/user-attachments/assets/f969349e-3c9b-4702-9e95-d0b7f230f9c9)

![Image](https://github.com/user-attachments/assets/674ddaf7-6acf-4321-8666-9b2eaca2dd18)

**Comment by TFCx** [comment]
I meant : replace the "special red case" by styling those roads into white thinner line segments.

> It's hard to tell the pedestrian cell isn't connected to the outside, at least for the way I happened to draw the boundary this time with the river

I might be missing something. Are all of thos equivalent : "special red zones" <=?=>  disconnected roads <=?=>  pedestrian zones. Even considering the user defined boundary ?

**Comment by XioNoX** [comment]
Another idea based on what you said. For pedestrian/red cells maybe it could be shown as a regular cell, BUT with a dashed red contour in "cell mode" or as dashed streets (of regular cell color) in "street mode" ?

**Comment by TFCx** [comment]
Bumping that issue after a friend's remark.

Both pedestrian streets and parking lots should have their display adjusted to that they are less of a distraction graphically. I would still advocate for _thinner_ lines like you demonstrated here:

![Image](https://github.com/user-attachments/assets/7400d8e5-ab28-4677-8a8b-7648d367daf5)
